### PR TITLE
feat: fixed followup and unassignment

### LINKED ISSUE #408: Unreliable Followups/Unassignment
The bot seems to be inconsistent on who it follows up with and then unassigns when dealing with bounties that are assigned but have no updates happening. This is an important feature that should be functional, or else we will have many bounties being reserved and neglected. 

I have a feeling that this stopped being reliable after implementing the capability for the bot to monitor linked pull requests for updates. But this feeling may be unfounded cause I haven't done the research to confirm it. 

              Thanks @pavlovcik. will complete this asap

_Originally posted by @sweetapplepumpkinpie in https://github.com/ubiquity/ubiquibot/issues/176#issuecomment-1583449657_
            

---

I will compile examples as I find them.

- https://github.com/ubiquity/ubiquity-dollar/issues/516

### PR DESCRIPTION
**Pull request by wannacfuture**
Resolves #408 

QA issue1: https://github.com/wannacfuture/Battleship/issues/20#issuecomment-1596167479
There, the user assigned that issue 5 days ago, but never did any activity. So the bot follows him up.
QA issue2: https://github.com/wannacfuture/Battleship/issues/17#issuecomment-1596165596
There, the use assigned that issue last week but never did any activity. So the bot releases the bounty

### _**IMPORTANT**_
Here, the bot monitors the activity of bounty hunter.
And last activity time means the latest time among:
- last comment time on the issue from the bounty hunter.
- committed time of last commit on the linked pull request from the bounty hunter.
- created time of last comment on the linked pull request from the bounty hunter.

Test successful.

### CODE REVIEW

**Comment by pavlovcik** [comment]
The linked examples look good but what was the root cause of the issue?

[> This bounty requires that you figure out the root cause of the issue, as well as provide a fix.](https://github.com/ubiquity/ubiquibot/issues/408#issuecomment-1594158464)

File: src/helpers/issue.ts [hunk 1]
```diff
@@ -374,6 +374,23 @@ export const getOpenedPullRequests = async (username: string) => {
   return prs.filter((pr) => !pr.draft && pr.user?.login === username);
 };
 
+export const getCommitsOnPullRequest = async (pull_number: number) => {
```
Comment (pavlovcik) [inline src/helpers/issue.ts:377]:
```suggestion
export const getCommitsOnPullRequest = async (pullRequest: number) => {
```

We use camel case!

**Review by pavlovcik — CHANGES_REQUESTED** [review]

**Comment by wannacfuture** [comment]
Yea, the root cause of the issue was that the mentioned comment of that bounty hunter is created less than a week before.
So the bot identified that issue as it has activity within the week so, it didn't unassign the hunter.

File: src/helpers/issue.ts (hunk 1 above)
Comment (wannacfuture) [reply src/helpers/issue.ts]: fixed.

**Comment by wannacfuture** [comment]
@0xcodercrane , could you review this one?

**Review by 0xcodercrane — APPROVED** [review]
The codebase looks good. QA is also good but @Draeieg needs to verify it finally.

**Comment by pavlovcik** [comment]
> QA is also good but @Draeieg needs to verify it finally.

Shouldn't you request a review from Draeieg if you need QA?

**Comment by wannacfuture** [comment]
@Draeieg ? 😳

**Review by Draeieg — APPROVED** [review]

**Comment by pavlovcik** [comment]
> Yea, the root cause of the issue was that the mentioned comment of that bounty hunter is created less than a week before.
> So the bot identified that issue as it has activity within the week so, it didn't unassign the hunter.

I don't believe that's correct. The reason why I made the issue is because it had already been a full week, and I expected that they should have been unassigned.

**Comment by wannacfuture** [comment]
> I don't believe that's correct. The reason why I made the issue is because it had already been a full week, and I expected that they should have been unassigned. @Draeieg do you have any theories?

No, it had already been a full week, but there is a comment of the hunter on the issue within the week before.
Could you take a closer look? @pavlovcik ?

**Comment by pavlovcik** [comment]
The easiest way to prove your point to me is if you can link the two comments you are talking about here.

**Comment by wannacfuture** [comment]
on the issue https://github.com/ubiquity/ubiquibot/issues/176#issue-1614068275 , 
the bot commented about the time limit on https://github.com/ubiquity/ubiquibot/issues/176#issuecomment-1581824756 .
You see? the created time is `Jun 8 2023 12:09 PM GMT+9`
then Lets move to this comment : https://github.com/ubiquity/ubiquibot/issues/176#issuecomment-1583449657
the created time is `Jun 9 2023 7:03 AM GMT+9`
Then, as you can see - @pavlovcik , you mentioned about that issue that it is not unassigned, but actually at that time - `Jun 16 2023 6:57 AM GMT +9` I think you can calculate the time - it has been less than a week!! right?
then the hunter added comment at `Jun 16 2023 5:44 PM GMT +9` (plz don't think the bot works real-time, lol)
if the hunter added comment the day after, or something the bot would have unassigned it but luckily it wasn't.

**Comment by pavlovcik** [comment]
The bot is supposed to follow up with a comment within four days. The comment examples you provided are more than four days apart.

**Comment by rndquu** [comment]
@0xcodercrane do we have [FOLLOW_UP_TIME](https://github.com/ubiquity/ubiquibot/blob/development/src/bindings/config.ts#L48) and [DISQUALIFY_TIME](https://github.com/ubiquity/ubiquibot/blob/development/src/bindings/config.ts#L49) env variables set in the netlify prod environment? If so then what are their values?

**Comment by wannacfuture** [comment]
@rndquu , The bot add comment : "Do you have any update ...?" after FOLLOW_UP_TIME has passed.
And the bot unassines the hunter after DISQUALIFY_TIME has passed.

Seems something went wrong at that moment(FOLLOW_UP_TIME has not been set or something).
I think we need to take a look for some period if my MR is working. @pavlovcik

**Comment by pavlovcik** [comment]
> @0xcodercrane do we have [FOLLOW_UP_TIME](https://github.com/ubiquity/ubiquibot/blob/development/src/bindings/config.ts#L48) and [DISQUALIFY_TIME](https://github.com/ubiquity/ubiquibot/blob/development/src/bindings/config.ts#L49) env variables set in the netlify prod environment? If so then what are their values?

You can verify in the future by logging into @ubiquity-bounties and then logging into netlify. 

### Production
image

### Staging
image

**Comment by rndquu** [comment]
> > @0xcodercrane do we have [FOLLOW_UP_TIME](https://github.com/ubiquity/ubiquibot/blob/development/src/bindings/config.ts#L48) and [DISQUALIFY_TIME](https://github.com/ubiquity/ubiquibot/blob/development/src/bindings/config.ts#L49) env variables set in the netlify prod environment? If so then what are their values?
> 
> * You can verify in the future by logging into @ubiquity-bounties and then logging into netlify.
> * Great diagnosis rndquu!
> 
> * I just set `FOLLOW_UP_TIME` to `4`
> * I just set `DISQUALIFY_TIME` to `7`
> 
> Does that mean that this pull request was unnecessary?

> You can verify in the future by logging into @ubiquity-bounties

You mean authenticate by login and password? I don't have a password for the @ubiquity-bounties account

> I just set `FOLLOW_UP_TIME` to `4`,  I just set `DISQUALIFY_TIME` to `7`

This breaks unassignment because [here](https://github.com/ubiquity/ubiquibot/blob/development/src/bindings/config.ts#L48) `ms` is used which converts `ms(4)` to `4ms` string value which breaks unassignment. You should either delete `FOLLOW_UP_TIME` and `DISQUALIFY_TIME` env variables either set them to their correct default [string values](https://github.com/ubiquity/ubiquibot/blob/development/src/configs/shared.ts#L24)

> Does that mean that this pull request was unnecessary?

No, this PR is necessary. I simply wanted to check whether env variables (`FOLLOW_UP_TIME` and `DISQUALIFY_TIME`) are set and that their values are correct.

**Comment by pavlovcik** [comment]
> You mean authenticate by login and password? I don't have a password for the @ubiquity-bounties account

Shared in direct message.

> This breaks unassignment because [here](https://github.com/ubiquity/ubiquibot/blob/development/src/bindings/config.ts#L48) ms is used which converts ms(4) to 4ms string value which breaks unassignment. You should either delete FOLLOW_UP_TIME and DISQUALIFY_TIME env variables (this way default values will be used) either set them to their correct default [string values](https://github.com/ubiquity/ubiquibot/blob/development/src/configs/shared.ts#L24)

Okay then try logging in and resetting the values to what you think makes sense.

**Comment by rndquu** [comment]
> > You mean authenticate by login and password? I don't have a password for the @ubiquity-bounties account
> 
> Shared in direct message.
> 
> > This breaks unassignment because [here](https://github.com/ubiquity/ubiquibot/blob/development/src/bindings/config.ts#L48) ms is used which converts ms(4) to 4ms string value which breaks unassignment. You should either delete FOLLOW_UP_TIME and DISQUALIFY_TIME env variables (this way default values will be used) either set them to their correct default [string values](https://github.com/ubiquity/ubiquibot/blob/development/src/configs/shared.ts#L24)
> 
> Okay then try logging in and resetting the values to what you think makes sense.

@0xcodercrane notice that I've updated env variables in prod netlify deploy to these values (which are used by [default](https://github.com/ubiquity/ubiquibot/blob/development/src/configs/shared.ts#L24) if `FOLLOW_UP_TIME` and `DISQUALIFY_TIME` are not set at all):
```
FOLLOW_UP_TIME="4 days"
DISQUALIFY_TIME="7 days"
```

@Draeieg FYI on staging we have the following "unassign" setup:
```
FOLLOW_UP_TIME="24 hrs"
DISQUALIFY_TIME="25 hrs"
```

**Comment by Draeieg** [comment]
> > > You mean authenticate by login and password? I don't have a password for the @ubiquity-bounties account
> > 
> > Shared in direct message.
> > 
> > > This breaks unassignment because [here](https://github.com/ubiquity/ubiquibot/blob/development/src/bindings/config.ts#L48) ms is used which converts ms(4) to 4ms string value which breaks unassignment. You should either delete FOLLOW_UP_TIME and DISQUALIFY_TIME env variables (this way default values will be used) either set them to their correct default [string values](https://github.com/ubiquity/ubiquibot/blob/development/src/configs/shared.ts#L24)
> > 
> > Okay then try logging in and resetting the values to what you think makes sense.
> 
> @0xcodercrane notice that I've updated env variables in prod netlify deploy to these values (which are used by [default](https://github.com/ubiquity/ubiquibot/blob/development/src/configs/shared.ts#L24) if `FOLLOW_UP_TIME` and `DISQUALIFY_TIME` are not set at all):
> ```
> FOLLOW_UP_TIME="4 days"
> DISQUALIFY_TIME="7 days"
> ```
> 
> @Draeieg FYI on staging we have the following "unassign" setup:
> ```
> FOLLOW_UP_TIME="24 hrs"
> DISQUALIFY_TIME="25 hrs"
> ```

Yes, I was the one who requested those times to facilitate QA

**Comment by Draeieg** [comment]
hey dm me at https://t.me/Draeieg  @wannacfuture 
7529d9aa-e610-4b30-8fbd-b948fd6ea3d7_text

**Comment by pavlovcik** [comment]
@Draeieg I can give you direct access to change environment variables if needed just dm me.

**Pull request merged**